src/GitHub.TeamFoundation.14/Services/VSGitServices.cs

```diff
@@ -39,6 +36,8 @@ public class VSGitServices : IVSGitServices
 
         [SuppressMessage("Microsoft.Performance", "CA1823:AvoidUnusedPrivateFields", Justification = "Used in VS2017")]
         readonly Lazy<IStatusBarNotificationService> statusBar;
+        [SuppressMessage("Microsoft.Performance", "CA1823:AvoidUnusedPrivateFields", Justification = "Used in VS2015")]
+        readonly Lazy<IVSServices> vsServices;
```

Comment:
Can you quickly explain this bit for my edification?

Reply:
There is a `TryOpenRepository` method on `IVSServices` that we only need to call when running on Visual Studio 2015. On Visual Studio 2017, the cloned repository automatically opens in folder view (which is new in Visual Studio 2017).

If we didn't call `TryOpenRepository`, you wouldn't see the URL change on `Team Explorer - Home` and there'd be no option to open a solution. You'd need to manually find and open the cloned solution.